app/services/the_tvdb_service.rb

```diff
@@ -1,41 +1,58 @@
 class TheTvdbService
+  class NotFound < StandardError; end
+
   BASE_URL = 'https://api.thetvdb.com'.freeze
   API_KEY = ENV['THE_TVDB_API_KEY'].freeze
 
-  # This will create new mappings that combine series/season together.
-  def create_tvdb_mapping!
-    return if mapping_data.blank?
+  def initialize(set_name)
+    @set_name = set_name
+  end
 
-    mapping_data['Anime'].each do |media_ids|
-      # set the media_id (this is for our database)
-      media_id = media_ids[0]
-      series_id = media_ids[1]['thetvdb/series']
-      season_id = media_ids[1]['thetvdb/season']
+  def import!
+    data = query_data
+    return if data.blank?
 
-      response = get(build_episode_path(series_id, season_id))
+    data['Anime'].each do |(anime_id, mappings)|
+      series_id, season_id = mappings['thetvdb'].split('/')
 
-      next if response.code == 404
-      raise 'TVDB Error' unless response.success?
+      response = get(build_episode_path(series_id, season_id))
+      next unless response
 
-      response = JSON.parse(response.body)
+      # update the episode count if it does not exist.
+      tvdb_episodes_count = response['data'].count
+      anime = update_anime_episode_count(anime_id, tvdb_episodes_count)
 
-      # Filters depending on if a airedSeasonID is present
-      response['data'] = season_filter(response['data'], season_id)
+      # we don't want to add their data if the episode counts don't match up
+      # TODO: we should log this because it would be good to double check.
+      next unless [anime.episode_count, anime.episode_count_guess].include?(tvdb_episodes_count)
 
-      season_number = find_season_number(response['data'])
-      create_mapping(media_id, series_id, season_number)
+      # creating/updating the episode
+      process_episode_data(response, anime_id, series_id)
     end
   end
 
-  # grabs all Mappings
-  def mapping_data
-    @md ||= Mapping.where(external_site: %w[thetvdb/series thetvdb/season])
-                   .pluck(:item_type, :item_id, :external_site, :external_id)
-                   .each_with_object({}) do |(item_type, item_id, external_site, external_id), acc|
-                     acc[item_type] ||= {}
-                     acc[item_type][item_id] ||= {}
-                     acc[item_type][item_id][external_site] = external_id
-                   end
+  def missing_thumbnails
+    return @mtd if @mtd
```

Comment:
Please don't use hngrn_ntn (hungarian notation) for variables.  Use whole words.

```ruby
return @missing_thumbnails if @missing_thumbnails
```